src/initialization/FGLinearization.h

```diff
@@ -19,25 +19,74 @@
 #ifndef FGLinearization_H_
 #define FGLinearization_H_
 
-#include "initialization/FGTrimmer.h"
 #include "math/FGStateSpace.h"
 #include <iomanip>
 #include <fstream>
 #include "models/FGAircraft.h"
 #include "models/propulsion/FGEngine.h"
 #include "models/propulsion/FGTurbine.h"
 #include "models/propulsion/FGTurboProp.h"
-#include "math/FGNelderMead.h"
 #include <stdexcept>
 #include <fstream>
 #include <cstdlib>
+#include <memory>
+#include <utility>
 
 namespace JSBSim {
 
+template <class T>
+using Vector2D = std::vector<std::vector<T>>;
+
+/** \brief Class used to create linear models from FGFDMExec instances.
+ */
 class FGLinearization
 {
+    Vector2D<double> A,B,C,D;
+    std::vector<double> x0, u0, y0;
+    std::vector<string> x_names, u_names, y_names, x_units, u_units, y_units;
+    std::string aircraft_name;
 public:
-    FGLinearization(FGFDMExec * fdmPtr, int mode);
+    /**
+     * @param fdmPtr Already configured FGFDMExec instance used to create the new linear model.
+     */
+    FGLinearization(FGFDMExec * fdmPtr);
+
+    /**
+     * Write Scicoslab source file with the state space model to a
+     * file in the current working directory.
+     */
+    void WriteScicoslab() const;
+
+    /**
+     * Write Scicoslab source file with the state space model to the given path.
+     *
+     * @param path
+     */
+    void WriteScicoslab(std::string& path) const;
+
+    /**
+     * Get the state space model matrices.
+     *
+     * @param A_ System matrix
+     * @param B_ Input matrix
+     * @param C_ Output matrix
+     * @param D_ Feedforward matrix
+     */
+    void GetStateSpace(Vector2D<double> & A_, Vector2D<double> & B_,
+                       Vector2D<double> & C_, Vector2D<double> & D_) const;
+
+    std::vector<double> GetInitialState() const;
+    std::vector<double> GetInitialInput() const;
+    std::vector<double> GetInitialOutput() const;
+
+    std::vector<std::string> GetStateNames() const;
+    std::vector<std::string> GetInputNames() const;
+    std::vector<std::string> GetOutputNames() const;
+
+    std::vector<std::string> GetStateUnits() const;
+    std::vector<std::string> GetInputUnits() const;
+    std::vector<std::string> GetOutputUnits() const;
```

Comment:
Trivial methods like that must be inlined in the header and return a const reference to avoid calling the copy constructor whenever possible (note that the `inline` keyword is not needed for the compiler to inline methods)
```suggestion
    const std::vector<double>& GetInitialState(void) const { return x0; }
    const std::vector<double>& GetInitialInput(void) const { return u0; }
    const std::vector<double>& GetInitialOutput(void) const { return y0; }

    const std::vector<std::string>& GetStateNames(void) const { return x_names; }
    const std::vector<std::string>& GetInputNames(void) const { return u_names; }
    const std::vector<std::string>& GetOutputNames(void) const { return y_names; }

    const std::vector<std::string>& GetStateUnits(void) const { return x_units; }
    const std::vector<std::string>& GetInputUnits(void) const { return u_units; }
    const std::vector<std::string>& GetOutputUnits(void) const { return y_units; } 
```